### PR TITLE
Remove unnecessary subclass of stdlib decimal.Decimal

### PR DESCRIPTION
**Pull request by jdufresne**
The Decimal class from stdlib provides the all required functionality.

### CODE REVIEW

**Comment by ofek** [comment]
It automatically handles floats correctly now?

**Comment by jdufresne** [comment]
> It automatically handles floats correctly now?

Yes, unless I'm misunderstanding a use case.

```pycon
>>> Decimal(3.14)
Decimal('3.140000000000000124344978758017532527446746826171875')
>>> Decimal(3.14) * 2
Decimal('6.280000000000000248689957516')
>>> Decimal(3.14) / 2
Decimal('1.570000000000000062172489379')
```

The test suite passes, as does the mypy type checker.

File: binary/core.py [hunk 1]
```diff
@@ -132,7 +131,9 @@ def convert_units(n, unit=BYTE, to=None, si=False, exact=False):
         raise ValueError(f'{unit} is not a valid binary unit.')
 
     # Always work with bytes to simplify logic.
-    n *= Decimal(unit) if exact else unit
+    if exact:
+        n = Decimal(n)
```
Comment (jdufresne) [inline binary/core.py]: @ofek 

If the concern is the floating point precision way off to the right, we can coerce this to a str first.

 ```suggestion
         n = Decimal(str(n))
```

If that is the concern, I can add a test for it.

**Comment by ofek** [comment]
yes

```
$ python -c "import sys;from decimal import Decimal as D;print(sys.version);print(D(0.1) + D(0.2));print(D('0.1') + D('0.2'))"
3.8.1 (tags/v3.8.1:1b293b6, Dec 18 2019, 23:11:46) [MSC v.1916 64 bit (AMD64)]
0.3000000000000000166533453694
0.3
```

**Comment by jdufresne** [comment]
Thanks I applied the fix I suggested above and added a suite of tests to show the same behavior before and after.

The test is available as a standalone change in #6.

**Comment by jdufresne** [comment]
Rebased

**Review by ofek — APPROVED** [review]

**Pull request merged**